// @aws-sdk/client-transcribe-streaming/dist-es/commands/StartMedicalScribeStreamCommand.js
import { getEventStreamPlugin } from "@aws-sdk/middleware-eventstream";
import { getTranscribeStreamingPlugin } from "@aws-sdk/middleware-sdk-transcribe-streaming";
import { getWebSocketPlugin } from "@aws-sdk/middleware-websocket";
import { getEndpointPlugin } from "@smithy/middleware-endpoint";
import { Command as $Command } from "@smithy/smithy-client";
import { commonParams } from "../endpoint/EndpointParameters";
import { StartMedicalScribeStream$ } from "../schemas/schemas_0";
export { $Command };
export class StartMedicalScribeStreamCommand extends $Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [
        getEndpointPlugin(config, Command.getEndpointParameterInstructions()),
        getEventStreamPlugin(config),
        getWebSocketPlugin(config, {
            headerPrefix: 'x-amzn-transcribe-',
        }),
        getTranscribeStreamingPlugin(config),
    ];
})
    .s("Transcribe", "StartMedicalScribeStream", {
    eventStream: {
        input: true,
        output: true,
    },
})
    .n("TranscribeStreamingClient", "StartMedicalScribeStreamCommand")
    .sc(StartMedicalScribeStream$)
    .build() {
}
